model/histogram/float_histogram.go

```diff
@@ -916,11 +930,14 @@ type allFloatBucketIterator struct {
 func (i *allFloatBucketIterator) Next() bool {
 	switch i.state {
 	case -1:
-		if i.negIter.Next() {
-			i.currBucket = i.negIter.At()
-			if i.currBucket.Upper > -i.h.ZeroThreshold {
+		if i.leftIter.Next() {
+			i.currBucket = i.leftIter.At()
+			if i.currBucket.Upper < 0 && i.currBucket.Upper > -i.h.ZeroThreshold {
 				i.currBucket.Upper = -i.h.ZeroThreshold
 			}
+			if i.currBucket.Lower > 0 && i.currBucket.Lower < i.h.ZeroThreshold {
+				i.currBucket.Lower = i.h.ZeroThreshold
+			}
```

Comment:
Since both conditions are mutually exclusive, it's more efficient _and_ easier to read if you use a switch here:
```suggestion
			switch {
			case i.currBucket.Upper < 0 && i.currBucket.Upper > -i.h.ZeroThreshold:
				i.currBucket.Upper = -i.h.ZeroThreshold
			case i.currBucket.Lower > 0 && i.currBucket.Lower < i.h.ZeroThreshold:
				i.currBucket.Lower = i.h.ZeroThreshold
			}
```

Same below for the `rightIter`.